content/docs/learn/typed-errors/from-either-to-raise.md

```diff
@@ -0,0 +1,260 @@
+should rather use exceptions (maybe combined with [resilience](../resilience/intro.md)).
+
+:::
+
+By the way, the Raise DSL provides several utility functions for common patterns.
+Checking an assertion and raising if false is one of those; so the most idiomatic
+way to write the code above is:
+
+```kotlin
+fun fooThatRaises(n: Int): Either<Error, String> = either {
+  ensure(n >= 0) { Error.NegativeInput }
+  val s = f(n).bind()
+  val t = g(s).bind()
+  t.summarize()
+}
+```
+
+## Transforming the error values
+
+The fact that both `f` and `g` share the same `Error` type is key for
+the previous code blocks to compile. In every `Raise` scope, we have one _single_
+error type which you can `raise` or `bind` in the whole block of code.
+If this is not the case, you have to _bridge_ the different error types, which
+is done using functions like `mapLeft`.
+
+```kotlin
+fun bar(n: Int): Either<Error, String> =
+  f(n).flatMap { s ->
+    h(s).mapLeft { 
+      it.toString()
+    }.map { t ->
+      t.summarize()
+    }
+  }
+```
+
+Within the `Raise` DSL, the corresponding function is called `withError`.
+Following the discussion above, the `withError` function creates a new
+scope in which the error type is different. The first argument to
+`withError` describes how to transform failure values from the inner
+scope into the error type of the outer scope.
+
+```kotlin
+fun bar(n: Int): Either<Error, String> = either {
+  val s = f(n).bind()
+  val t = withError({ boo -> boo.toError() }) { h(s).bind() }
+  t.summarize()
+}
+```
+
+## More than one failure
+
+Sometimes you have not one, but many values for which you
+want to run a potentially-failing computation. Almost every library
+provides an "effectful map" (usually called `traverse`) to cover those cases.
+
+```kotlin
+fun foos(xs: List<Int>) = xs.traverse { foo(it) }
+```
+
+Another advantage of the `Raise` DSL is that you don't need all of those
+new combinators. The regular `map` on lists is enough — although you
+need to remember to `bind` the values to "inject" the errors.
+
+```kotlin
+fun foos(xs: List<Int>) = either {
+  xs.map { foo(it).bind() }
+  // alternatively
+  xs.map { foo(it) }.bindAll()
+}
+```
+
+This advantage is bigger than just dropping a few function names. Since
+you don't need specific effectful combinators, you are free to use any
+function operating on collections of elements, without restriction.
+
+:::tip Error accumulation
+
+By default `Raise` operates on a fail-first basis: once a failure is `raise`d,
+the whole block comes to a halt. If you are operating on a collection, this
+means that only the first error is reported. If you need to _accumulate_
+all errors that arise in some block, you need to switch to `mapOrAccumulate`.
+More information can be found in the 
+[Working with typed errors](working-with-typed-errors.md#accumulating-errors) page.
+
+:::
+
+## `Either` and `bind` no more
```

Comment:
Is it worth mentioning alternatives when wanting to adopt this style for methods with an existing receiver? (even if just highlighting the conflict)

i.e. if we had `fun Thing.summarize(): Either<Error, String>` and wanted to avoid the `Either` return type. (obviously context receivers is one option, heavily caveated by it being experimental)

Reply:
Done!